### PR TITLE
Fix the way addon-manager handles non-namespaced objects

### LINKED ISSUE #28451: Infinite retries of PersistentVolume POSTs
During all our tests we see one PV POST that succeeds:
```
I0704 03:06:46.999538       5 handlers.go:165] POST /api/v1/persistentvolumes: (139.86986ms) 201 [[kubectl/v1.3.0 (linux/amd64) kubernetes/caf9a4d] 127.0.0.1:36472]
```
All consecutive ones return 409 and are retried every 10s:
```
I0704 03:08:45.698643       5 handlers.go:165] POST /api/v1/persistentvolumes: (114.515217ms) 409 [[kubectl/v1.3.0 (linux/amd64) kubernetes/caf9a4d] 127.0.0.1:36942]
...
I0704 03:08:55.917765       5 handlers.go:165] POST /api/v1/persistentvolumes: (124.657052ms) 409 [[kubectl/v1.3.0 (linux/amd64) kubernetes/caf9a4d] 127.0.0.1:36962]
...
```
Some runs of our scalability tests are failing on PV POST latencies.

cc @wojtek-t @fgrzadkowski @saad-ali

### PR DESCRIPTION
**Pull request by gmarek**
Fixes #28451

cc @wojtek-t

### CODE REVIEW

**Comment by gmarek** [comment]
@k8s-bot e2e test this issue: #IGNORE
Node failed.

File: cluster/saltbase/salt/kube-addons/kube-addon-manager.yaml [hunk 1]
```diff
@@ -10,7 +10,7 @@ spec:
   hostNetwork: true
   containers:
   - name: kube-addon-manager
-    image: gcr.io/google-containers/kube-addon-manager:v5
```
Comment (luxas) [inline cluster/saltbase/salt/kube-addons/kube-addon-manager.yaml]: The same version bump should also be done in cluster/images/hyperkube/static-pods/addon-manager.
Can you also add a note to this file about that, so it becomes easier to notice?

**Comment by gmarek** [comment]
@k8s-bot smoke test this issue: #IGNORE
cluster startup failure

**Comment by gmarek** [comment]
@k8s-bot smoke test this issue: #IGNORE
cluster startup failure for smoke tests

**Comment by gmarek** [comment]
Ping @mikedanese

**Comment by fgrzadkowski** [comment]
LGTM

File: cluster/addons/addon-manager/kube-addon-update.sh [hunk 2]
```diff
@@ -262,7 +260,11 @@ function create-object() {
     log INFO "Creating new ${obj_type} from file ${file_path} in namespace ${namespace}, name: ${obj_name}"
     # this will keep on failing if the ${file_path} disappeared in the meantime.
     # Do not use too many retries.
-    run-until-success "${KUBECTL} create --namespace=${namespace} -f ${file_path}" ${NUM_TRIES} ${DELAY_AFTER_ERROR_SEC}
+    if [[ -n "${namespace}" ]]; then
+        run-until-success "${KUBECTL} create --namespace=${namespace} -f ${file_path}" ${NUM_TRIES} ${DELAY_AFTER_ERROR_SEC}
```
Comment (mikedanese) [inline cluster/addons/addon-manager/kube-addon-update.sh]: If you create an object with out --namespace and it declares a non default namespace in the yml, does it use the namespace from the yml? If so (which I thought was the case) you could just remove the namespace arg

**Pull request merged**